compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -248,10 +248,12 @@ pub fn minimal_passes() -> Vec<SsaPass<'static>> {
         // which was called in the AST not being called in the SSA. Such functions would cause
         // panics later, when we are looking for global allocations.
         SsaPass::new(Ssa::remove_unreachable_functions, "Removing Unreachable Functions"),
-        // We need a DIE pass to populate `used_globals`, otherwise it will panic later.
-        SsaPass::new(Ssa::dead_instruction_elimination, "Dead Instruction Elimination"),
         // We need to add an offset to constant array indices in Brillig.
+        // This can change which globals are used, because constant creation might result
+        // in the (re)use of otherwise unused global values.
         SsaPass::new(Ssa::brillig_array_get_and_set, "Brillig Array Get and Set Optimizations"),
+        // We need a DIE pass to populate `used_globals`, otherwise it will panic later.
```

Comment:
Could we add these comments to `brillig_array_get_and_set` in the primary passes as well? I do think I put DIE after `brillig_array_get_and_set` in the primary passes due to this reason as well, but it was an oversight to not leave any comments.